metadata/src/main/java/org/apache/kafka/controller/metrics/ControllerMetricsChanges.java

```diff
@@ -74,21 +80,38 @@ public int partitionsWithoutPreferredLeaderChange() {
         return partitionsWithoutPreferredLeaderChange;
     }
 
-    void handleBrokerChange(BrokerRegistration prev, BrokerRegistration next) {
+    void handleBrokerChange(BrokerRegistration prev, BrokerRegistration next, ControllerMetadataMetrics metrics) {
         boolean wasFenced = false;
         boolean wasActive = false;
+        boolean wasInControlledShutdown = false;
         if (prev != null) {
             wasFenced = prev.fenced();
             wasActive = !prev.fenced();
+            wasInControlledShutdown = prev.inControlledShutdown();
+        } else {
+            metrics.addBrokerRegistrationStateMetric(next.id());
         }
         boolean isFenced = false;
         boolean isActive = false;
+        boolean isInControlledShutdown = false;
         if (next != null) {
             isFenced = next.fenced();
             isActive = !next.fenced();
+            isInControlledShutdown = next.inControlledShutdown();
+            if (isFenced) {
```

Comment:
we should be able to just pass the `BrokerRegistration` object to `setBrokerRegistrationState`, right?